### PR TITLE
Create album reviews endpoint

### LINKED ISSUE #74: Create endpoint for adding album reviews
Create endpoint on backend to add reviews to track.

endpoint:

`/albums/{album id on spotify}/reviews`

- POST
   1. It will receive a JSON with the format:
    ```json
   {
     "user": "userId",
     "album": "id of the album on spotify",
     "rating": 4.5,
     "review": "[OPTIONAL OBJECT TO BE DEFINED]"
    }
   ```
   1. Store it in a collection "reviews/albums" (there is a concept of subcollection on firestore) using the [firebase cloud firestore](https://firebase.google.com/docs/firestore), we already have an example using the cloud firestore [here](https://github.com/calluswhatyouwant/musicritic/blob/master/server/src/users/userCollections.js)
   1. Return the review JSON with its id

if the #69 is already done, this endpoint must be protected to be called only by logged users

### PR DESCRIPTION
**Pull request by paulojbleitao**
This PR creates the album reviews endpoint, which has the ability to retrieve, create and update album reviews. It functions basically the same as the track reviews endpoint.

I also refactored the `trackReviewCollections.js` file (now called just `reviewCollections.js`) so that it can also be used for the album operations, since they have the same logic. Most importantly, I changed the `TrackReviewModel` (now `ReviewModel`) a bit. Previously, it looked like this:

```js
type TrackReviewModel = {
    id: string | null,
    authorUid: string,
    trackId: string,
    rating: number,
    review: string | null,
};
```

Now, it has become:

```js
type ReviewModel = {
    id: string | null,
    authorUid: string,
    contentId: string,
    rating: number,
    review: string | null,
    contentType: 'track' | 'album',
};
```

In other words, `trackId` was renamed to `contentId` since it doesn't relate only to tracks anymore; and there's a new `contentType` field so we can differentiate between tracks and albums.

This PR resolves #74

### CODE REVIEW

File: server/src/reviews/reviewCollections.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    const ref = Reviews[review.contentType].doc();
+    review.id = ref.id;
+    await ref.set(review);
+    return review;
+};
+
+export const getUserReview = (
+    contentId: string,
+    authorUid: string,
+    contentType: string
```
Comment (JoseRenan) [inline server/src/reviews/reviewCollections.js:45]: I think is better to put the type as `"album" | "track"`

File: server/src/reviews/albumReviewController.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    const albumId = req.params.albumId;
+    getReviews(albumId, 'album')
+        .then(reviews => res.status(200).send(reviews))
+        .catch(error => res.status(error.status).send(error));
+});
+
+router.get('/album/:albumId/reviews/me', checkAuth, (req, res) => {
+    const albumId = req.params.albumId;
+    const authorUid = req.user.uid;
+    getUserReview(albumId, authorUid, 'album')
```
Comment (JoseRenan) [inline server/src/reviews/albumReviewController.js:24]: I think it's better to create a const for album and track strings, to be reused in all params

**Review by JoseRenan — COMMENTED** [review]
Added minor issues just to improve the code "understandability" if this word exists :laughing:

File: server/src/reviews/albumReviewController.js (hunk 2 above)
Comment (paulojbleitao) [reply server/src/reviews/albumReviewController.js:24]: I agree, though I'm not sure where I'd put them :thinking: Since they should be used across a few files. Should I make a consts file or something like that?

Comment (JoseRenan) [reply server/src/reviews/albumReviewController.js:24]: I usually prefer to start putting where it's being used now (in this case, albumReviewController or albumReviewCollection), and when other modules start using it too we decide if it's better to put it in another place

**Comment by JoseRenan** [comment]
@paulojbleitao the GET /track/:id/reviews/me is not working properly, could you take a look? I think is missing a .data() in the return :thinking:

**Review by JoseRenan — APPROVED** [review]

**Review by JRobsonJr — APPROVED** [review]
